crates/sui-framework/packages/deepbook/sources/clob_v2.move

```diff
@@ -1010,6 +1005,7 @@ module deepbook::clob_v2 {
         // Match the bid order against the asks Critbit Tree in the same way as a market order but up until the price level found in the previous step.
         // If the bid order is not completely filled, inject the remaining quantity to the bids Critbit Tree according to the input price and order id.
         // If limit ask order, vice versa.
+        assert!(self_matching_prevention == CANCEL_OLDEST, EInvalidSelfMatchingPreventionArg);
```

Comment:
- Will more self matching prevention modes be added in the future?
- For now, this is not needed, right (because the only implemented logic is `CANCEL_OLDEST`?